katdal/test/test_chunkstore_s3.py

```diff
@@ -0,0 +1,103 @@
+################################################################################
+# Copyright (c) 2011-2016, National Research Foundation (Square Kilometre Array)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Tests for :py:mod:`katdal.chunkstore_s3`."""
+
+import tempfile
+import shutil
+import subprocess
+import time
+
+import numpy as np
+from numpy.testing import assert_array_equal
+from nose import SkipTest
+from nose.tools import assert_raises
+import requests
+
+import katdal.chunkstore_s3
+from katdal.chunkstore_s3 import S3ChunkStore
+
+
+class TestS3ChunkStore(object):
+    def setup(self):
+        """Start Fake S3 service running on temp dir, and ChunkStore on that."""
+        self.tempdir = tempfile.mkdtemp()
+        host = 'localhost'
+        # Choose a random port and hope it's available
+        port = np.random.randint(10000, 60000)
+        url = "http://%s:%d" % (host, port)
+        try:
+            self.fakes3 = subprocess.Popen(['fakes3', 'server', '-H', host,
+                                            '-p', str(port), '-r', self.tempdir])
+        except OSError:
+            self.fakes3 = None
+            self.teardown()
+            raise SkipTest('Could not start fakes3 server (is it installed?)')
+        start = time.time()
+        # Give up after waiting a few seconds for Fake S3
+        while time.time() - start <= 10:
+            try:
+                response = requests.get(url, timeout=0.5)
+            except requests.exceptions.ConnectionError:
+                time.sleep(0.3)
+            else:
+                # In case there is already another service on this port (even S3)
+                if response.reason == 'OK':
+                    break
```

Comment:
This seems like a very weak test. Does fakes3 return any sort of HTTP header to advertise itself, or is there a request one can do to get the version?

Reply:
It is a very weak test. It basically disambiguates FakeS3 from Minio and other "real" S3s. I'll see if there is a slightly stronger test that does not suck up another week of my time.

Reply:
It looks like FakeS3 starts up with a FakeS3 bucket: if I query it, I get
```
<?xml version="1.0" encoding="UTF-8"?><ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Owner><ID>123</ID><DisplayName>FakeS3</DisplayName></Owner><Buckets></Buckets></ListAllMyBucketsResult>
```

Reply:
Thanks - that looks promising.

Reply:
Removed this test completely.